src/aws/auth.rs

```diff
@@ -133,13 +150,37 @@ impl AwsAuthentication {
             Self::AccessKey {
                 access_key_id,
                 secret_access_key,
-            } => Ok(SharedCredentialsProvider::new(Credentials::from_keys(
-                access_key_id.inner(),
-                secret_access_key.inner(),
-                None,
-            ))),
-            AwsAuthentication::File { .. } => {
-                Err("Overriding the credentials file is not supported.".into())
+                assume_role,
+                region,
+            } => {
+                let provider = SharedCredentialsProvider::new(Credentials::from_keys(
+                    access_key_id.inner(),
+                    secret_access_key.inner(),
+                    None,
+                ));
+                if let Some(assume_role) = assume_role {
+                    let auth_region = region.clone().map(Region::new).unwrap_or(service_region);
+                    let provider = AssumeRoleProviderBuilder::new(assume_role)
+                        .region(auth_region)
+                        .build(provider);
+                    return Ok(SharedCredentialsProvider::new(provider))
+                }
+                Ok(provider)
+            },
+            AwsAuthentication::File {
+                credentials_file,
+                profile,
+            } => {
+                let profile_files = ProfileFiles::builder()
+                    .with_file(ProfileFileKind::Credentials, credentials_file)
+                    .build();
+                let mut builder = ProfileFileCredentialsProvider::builder()
+                    .profile_files(profile_files);
+                if let Some(profile) = profile {
+                    builder = builder.profile_name(profile);
+                }
+                let provider = builder.build();
+                Ok(SharedCredentialsProvider::new(provider))
```

Comment:
Merging in `master` will conflict here, the code on master should be able to replace all of this - I think.